packages/web-awesome/src/components/Tree/Tree.tsx

```diff
@@ -16,9 +17,13 @@ interface TreeProps {
 }
 
 const Tree: FunctionComponent<TreeProps> = ({ tree, statusFilter, root, name, statistic }) => {
-  const [isOpened, setIsOpen] = useState(statistic === undefined || !!statistic.failed || !!statistic.broken);
-  const toggleTree = () => {
+  const isEarlyCollapsed = collapsedTrees.value.has(tree.nodeId);
+  const haveFailedSteps = statistic === undefined || !!statistic?.failed || !!statistic?.broken;
+  const [isOpened, setIsOpen] = useState(() => (isEarlyCollapsed ? !haveFailedSteps : Boolean(haveFailedSteps)));
```

Comment:
Again, why do we need to wrap `haveFailedSteps`? As far as I see, it's `boolean` in any case

Reply:
Don't we need to keep it closed/opened independently of failed steps inside? If the user collapsed the tree, it's ok to keep the tree as is without additional logic.